client/SMCE_Client.cpp

```diff
@@ -0,0 +1,339 @@
+#include <utility>                                                                                                               
+#include <string>                                                                                                                
+#include <vector>
+#include <termcolor.hpp>  
+#include <lyra.hpp>
+#include <UartToFile.hpp>                                                                                                                   
+#include <SMCE/Board.hpp>                                                                                                        
+#include <SMCE/BoardConf.hpp>                                                                                                    
+#include <SMCE/BoardView.hpp>                                                                                                    
+#include <SMCE/Sketch.hpp>                                                                                                       
+#include <SMCE/SketchConf.hpp>                                                                                                   
+#include <SMCE/Toolchain.hpp>
+                                                                                               
+                                                                                                                                 
+using namespace std::literals;                                                                                                   
+                                                                                                                                                                                                                                                                                                    
+void print_help(const char* argv0){                                                                                              
+    std::cout << "Usage: " << argv0 << " <fully-qualified-board-name> <path-to-sketch>" << std::endl;                            
+}                                                                                                                                
+                                                                                                                                 
+void exit_sketch(int code){                                                                                                      
+    std::cerr << "Error code:  "<< code << std::endl;                                                                            
+}                                                                                                                                
+
+// atomic bool for handling threads
+static std::atomic_bool run_threads = true;
+static std::atomic_bool mute_uart = false;
+static std::mutex t_lock;
```

Comment:
AAAAAAH GLOBALS!! FLEE!!!
Seriously tho, globals are the root of all evil. Do not use globals.